### PR TITLE
CFE-3603: Added example illustrating inherit_from (master)

### PR DESCRIPTION
**Pull request by nickanderson**
New example, intended for use in docs.

### CODE REVIEW

**Review by craigcomstock — COMMENTED** [review]
I wonder about the length of the body names, maybe foo and bar would be easier to read? I understand the meaning but it's a bit hard to "see" at a glance with such large names.

Maybe make the simplest example you can first, and then a more advanced example like what you have here second?

File: examples/inherit_from_classes.cf [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        classes => my_set_some_extra_fancy_classes( "$(this.promiser)",
+                                                    "$(this.handle)",
+                                                    "some_class_to_handle_dependencies" );
+
+  reports:
+      "Defined classes:$(const.n)$(with)"
+        with => join( "$(const.n)", sort( classesmatching( "some_.*"), "lex" ));
+}
+
+body classes my_set_some_extra_fancy_classes( x, y, z)
```
Comment (olehermanse) [inline examples/inherit_from_classes.cf:25]:
```suggestion
body classes my_set_some_extra_fancy_classes(x, y, z)
```

File: examples/inherit_from_classes.cf [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# @brief In addition to the classes set by `set_some_fancy_classes` define `z` when the promise is repaired
+{
+        inherit_from => set_some_fancy_classes( $(x), $(y) );
+        promise_repaired => { "some_fancy_class_${x}_${y}_repaired", $(z) };
+}
+
+body classes set_some_fancy_classes ( x, y)
```
Comment (olehermanse) [inline examples/inherit_from_classes.cf:32]:
```suggestion
body classes set_some_fancy_classes(x, y)
```

**Comment by nickanderson** [comment]
Replaced by #4635 (i was targeting the wrong ticket number here).

**Pull request closed without merging**